File: app/src/main/java/com/overswayit/plesnisavezsrbije/viewmodels/ClubViewModelFactory.java
package com.overswayit.plesnisavezsrbije.viewmodels;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

/**
 * Created by lazarristic on 30/04/2019.
 * Copyright (c) 2019 PlesniSavezSrbije. All rights reserved.
 */
public class ClubViewModelFactory implements ViewModelProvider.Factory {

    private Application application;
    private Integer id;

    public ClubViewModelFactory(Application application, int id) {
        this.application = application;
        this.id = id;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new ClubViewModel(id, application);
    }
}

File: app/src/main/java/com/overswayit/plesnisavezsrbije/viewmodels/ClubViewModelFactory.kt
package com.overswayit.plesnisavezsrbije.viewmodels

import android.app.Application
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

/**
 * Created by lazarristic on 30/04/2019.
 * Copyright (c) 2019 PlesniSavezSrbije. All rights reserved.
 */
class ClubViewModelFactory(private val application: Application, private val id: Int?) : ViewModelProvider.Factory {

    @Suppress("UNCHECKED_CAST")
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return ClubViewModel(id!!, application) as T
    }
}
